apps/studio/components/interfaces/APIKeys/APIKeyRow.tsx

```diff
@@ -0,0 +1,193 @@
+  apiKey,
+}: {
+  apiKey: Extract<APIKeysData[number], { type: 'secret' | 'publishable' }>
+}) => {
+  const MotionTableRow = motion(TableRow)
+
+  return (
+    <MotionTableRow
+      layout
+      initial={{ opacity: 0, height: 0 }}
+      animate={{ opacity: 1, height: 'auto' }}
+      exit={{ opacity: 0, height: 0 }}
+      transition={{
+        type: 'spring',
+        stiffness: 500,
+        damping: 50,
+        mass: 1,
+      }}
+    >
+      <TableCell className="py-2">{apiKey.description || '/'}</TableCell>
+      <TableCell className="py-2">
+        <div className="flex flex-row gap-2">
+          <Input apiKey={apiKey} />
+        </div>
+      </TableCell>
+
+      <TableCell className="flex justify-end">
+        <DropdownMenu>
+          <DropdownMenuTrigger className="px-1 focus-visible:outline-none" asChild>
+            <Button
+              type="text"
+              size="tiny"
+              icon={
+                <MoreVertical size="14" className="text-foreground-light hover:text-foreground" />
+              }
+            />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent className="max-w-40" align="end">
+            <APIKeyDeleteDialog apiKey={apiKey} />
+            <DropdownMenuSeparator />
+            <APIKeyDeleteDialog apiKey={apiKey} />
+          </DropdownMenuContent>
+        </DropdownMenu>
+      </TableCell>
+    </MotionTableRow>
+  )
+}
+
+function Input({
+  apiKey,
+}: {
+  apiKey: Extract<APIKeysData[number], { type: 'secret' | 'publishable' }>
+}) {
+  const queryClient = useQueryClient()
+  const { ref: projectRef } = useParams()
+
+  const [show, setShowState] = useState(false)
+  const canReadAPIKeys = useCheckPermissions(PermissionAction.TENANT_SQL_ADMIN_WRITE, '*')
+
+  const isSecret = apiKey.type === 'secret'
+
+  const {
+    data,
+    isLoading: isLoadingApiKey,
+    error,
+    refetch: refetchApiKey,
+  } = useAPIKeyIdQuery(
+    {
+      projectRef,
+      id: apiKey.id as string,
+      reveal: true,
+    },
+    {
+      enabled: show,
+      staleTime: 0, // Data is considered stale immediately
+      cacheTime: 0, // Cache is cleared immediately after query becomes stale
+    }
+  )
+
+  async function onSubmitShow() {
+    setShowState(true)
+    // Set a timeout to invalidate the cache after a certain amount of time
+    setTimeout(() => {
+      setShowState(false)
+      queryClient.removeQueries({
+        queryKey: apiKeysKeys.single(projectRef, apiKey.id as string),
+        exact: true,
+      })
+    }, 10000) // Destroy query after 10 seconds
+  }
+
+  async function onCopy() {
+    // if ID already exists from a reveal action, return that
+    if (data?.api_key) return data?.api_key
+
+    try {
+      // fetch ID and then destroy query immediately
+      const result = await refetchApiKey()
+      queryClient.removeQueries({
+        queryKey: apiKeysKeys.single(projectRef, apiKey.id as string),
+        exact: true,
+      })
```

Comment:
likewise why do we need to destroy the query here?

Reply:
so we don't have API secrets stored in cache